protocol/contracts/amo/TpfAmo.sol

```diff
@@ -0,0 +1,547 @@
+pragma solidity ^0.8.4;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./interfaces/AMO__IPoolHelper.sol";
+import "./interfaces/AMO__IAuraBooster.sol";
+import "./interfaces/AMO__ITempleERC20Token.sol";
+import "./helpers/AMOCommon.sol";
+import "./interfaces/AMO__IAuraStaking.sol";
+
+/**
+ * @title AMO built for 50TEMPLE-50BB-A-USD balancer pool
+ *
+ * @dev It has a  convergent price to which it trends called the TPF (Treasury Price Floor).
+ * In order to accomplish this when the price is below the TPF it will single side withdraw 
+ * BPTs into TEMPLE and burn them and if the price is above the TPF it will 
+ * single side deposit TEMPLE and increase its BPT position that way. Alternatively
+ */
+contract TpfAmo is Ownable {
+    using SafeERC20 for IERC20;
+
+    AMO__IBalancerVault public immutable balancerVault;
+    // @notice BPT token address
+    IERC20 public immutable bptToken;
+    // @notice Aura booster
+    AMO__IAuraBooster public immutable booster;
+    // @notice pool helper contract
+    AMO__IPoolHelper public poolHelper;
```

Comment:
Does this need to be an external contract? Couldn't it just be an (internal) library that we can utilise?

Reply:
previously it was a library, also tried an abstract contract. code size limit was exceeded